## Final Project Submission

Please fill out:
* Student name: Sophia Mumbi Mbataru
* Student pace: Full time
* Scheduled project review date/time: 
* Instructor name: 
* Blog post URL:


# Business Problem
Microsoft sees all the big companies creating original video content and they want to get in on the fun. They have decided to create a new movie studio, but they don’t know anything about creating movies. You are charged with exploring what types of films are currently doing the best at the box office. You must then translate those findings into actionable insights that the head of Microsoft's new movie studio can use to help decide what type of films to create.



# DESCRIBING THE QUESTIONS
1. Are movies making money?
2. Which movie genre is most profitable?
3. How does the choice of the director, and/or Actor have on movie grossing?
4. What is the relationship/correlation between a movie's popularity and its release date? 


# Data Preparation

Data preparation helps prepare data from different datasets and transform them to clean data that will be used for data analysis to answer the questions.

In [1]:
#import standard python libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

# Loading data

In [2]:

bom_movies = pd.read_csv("zippedData/bom.movie_gross.csv")
bom_movies

title      studio  domestic_gross  \
0                                     Toy Story 3          BV     415000000.0   
1                      Alice in Wonderland (2010)          BV     334200000.0   
2     Harry Potter and the Deathly Hallows Part 1          WB     296000000.0   
3                                       Inception          WB     292600000.0   
4                             Shrek Forever After        P/DW     238700000.0   
...                                           ...         ...             ...   
3382                                    The Quake       Magn.          6200.0   
3383                  Edward II (2018 re-release)          FM          4800.0   
3384                                     El Pacto        Sony          2500.0   
3385                                     The Swan  Synergetic          2400.0   
3386                            An Actor Prepares       Grav.          1700.0   

     foreign_gross  year  
0        652000000  2010  
1        691300000  2010  
2        664300000  2010  
3        535700000  2010  
4        513900000  2010  
...            ...   ...  
3382           NaN  2018  
3383           NaN  2018  
3384           NaN  2018  
3385           NaN  2018  
3386           NaN  2018  

[3387 rows x 5 columns]

In [3]:
tmdb_movies = pd.read_csv("zippedData/tmdb.movies.csv")
tmdb_movies

Unnamed: 0            genre_ids      id original_language  \
0               0      [12, 14, 10751]   12444                en   
1               1  [14, 12, 16, 10751]   10191                en   
2               2        [12, 28, 878]   10138                en   
3               3      [16, 35, 10751]     862                en   
4               4        [28, 878, 12]   27205                en   
...           ...                  ...     ...               ...   
26512       26512             [27, 18]  488143                en   
26513       26513             [18, 53]  485975                en   
26514       26514         [14, 28, 12]  381231                en   
26515       26515      [10751, 12, 28]  366854                en   
26516       26516             [53, 27]  309885                en   

                                     original_title  popularity release_date  \
0      Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                          How to Train Your Dragon      28.734   2010-03-26   
2                                        Iron Man 2      28.515   2010-05-07   
3                                         Toy Story      28.005   1995-11-22   
4                                         Inception      27.920   2010-07-16   
...                                             ...         ...          ...   
26512                         Laboratory Conditions       0.600   2018-10-13   
26513                               _EXHIBIT_84xxx_       0.600   2018-05-01   
26514                                  The Last One       0.600   2018-10-01   
26515                                  Trailer Made       0.600   2018-06-22   
26516                                    The Church       0.600   2018-10-05   

                                              title  vote_average  vote_count  
0      Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                          How to Train Your Dragon           7.7        7610  
2                                        Iron Man 2           6.8       12368  
3                                         Toy Story           7.9       10174  
4                                         Inception           8.3       22186  
...                                             ...           ...         ...  
26512                         Laboratory Conditions           0.0           1  
26513                               _EXHIBIT_84xxx_           0.0           1  
26514                                  The Last One           0.0           1  
26515                                  Trailer Made           0.0           1  
26516                                    The Church           0.0           1  

[26517 rows x 10 columns]

In [4]:
tn_movie_budgets = pd.read_csv("zippedData/tn.movie_budgets.csv")
tn_movie_budgets

id  release_date                                        movie  \
0      1  Dec 18, 2009                                       Avatar   
1      2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2      3   Jun 7, 2019                                 Dark Phoenix   
3      4   May 1, 2015                      Avengers: Age of Ultron   
4      5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
...   ..           ...                                          ...   
5777  78  Dec 31, 2018                                       Red 11   
5778  79   Apr 2, 1999                                    Following   
5779  80  Jul 13, 2005                Return to the Land of Wonders   
5780  81  Sep 29, 2015                         A Plague So Pleasant   
5781  82   Aug 5, 2005                            My Date With Drew   

     production_budget domestic_gross worldwide_gross  
0         $425,000,000   $760,507,625  $2,776,345,279  
1         $410,600,000   $241,063,875  $1,045,663,875  
2         $350,000,000    $42,762,350    $149,762,350  
3         $330,600,000   $459,005,868  $1,403,013,963  
4         $317,000,000   $620,181,382  $1,316,721,747  
...                ...            ...             ...  
5777            $7,000             $0              $0  
5778            $6,000        $48,482        $240,495  
5779            $5,000         $1,338          $1,338  
5780            $1,400             $0              $0  
5781            $1,100       $181,041        $181,041  

[5782 rows x 6 columns]

# Data Understanding

Data Understanding helps us know where the data came from and how they relate tot he data anaysis questions
It helps us know what and who is in the data presented, and the properties of the variables to be used (Profit/ROI, budget and studio)

In [7]:
#Get the statistical description of the data

bom_movies.describe(include = 'all')

title studio  domestic_gross foreign_gross         year
count        3387   3382    3.359000e+03          2037  3387.000000
unique       3386    257             NaN          1204          NaN
top     Bluebeard    IFC             NaN       1200000          NaN
freq            2    166             NaN            23          NaN
mean          NaN    NaN    2.874585e+07           NaN  2013.958075
std           NaN    NaN    6.698250e+07           NaN     2.478141
min           NaN    NaN    1.000000e+02           NaN  2010.000000
25%           NaN    NaN    1.200000e+05           NaN  2012.000000
50%           NaN    NaN    1.400000e+06           NaN  2014.000000
75%           NaN    NaN    2.790000e+07           NaN  2016.000000
max           NaN    NaN    9.367000e+08           NaN  2018.000000

In [12]:
tmdb_movies.describe(include = 'all')

Unnamed: 0 genre_ids             id original_language original_title  \
count   26517.00000     26517   26517.000000             26517          26517   
unique          NaN      2477            NaN                76          24835   
top             NaN      [99]            NaN                en           Eden   
freq            NaN      3700            NaN             23291              7   
mean    13258.00000       NaN  295050.153260               NaN            NaN   
std      7654.94288       NaN  153661.615648               NaN            NaN   
min         0.00000       NaN      27.000000               NaN            NaN   
25%      6629.00000       NaN  157851.000000               NaN            NaN   
50%     13258.00000       NaN  309581.000000               NaN            NaN   
75%     19887.00000       NaN  419542.000000               NaN            NaN   
max     26516.00000       NaN  608444.000000               NaN            NaN   

          popularity release_date  title  vote_average    vote_count  
count   26517.000000        26517  26517  26517.000000  26517.000000  
unique           NaN         3433  24688           NaN           NaN  
top              NaN   2010-01-01   Eden           NaN           NaN  
freq             NaN          269      7           NaN           NaN  
mean        3.130912          NaN    NaN      5.991281    194.224837  
std         4.355229          NaN    NaN      1.852946    960.961095  
min         0.600000          NaN    NaN      0.000000      1.000000  
25%         0.600000          NaN    NaN      5.000000      2.000000  
50%         1.374000          NaN    NaN      6.000000      5.000000  
75%         3.694000          NaN    NaN      7.000000     28.000000  
max        80.773000          NaN    NaN     10.000000  22186.000000

In [51]:
tn_movie_budgets.describe(include = 'all')

id  release_date      movie production_budget domestic_gross  \
count   5782.000000          5782       5782              5782           5782   
unique          NaN          2418       5698               509           5164   
top             NaN  Dec 31, 2014  Halloween       $20,000,000             $0   
freq            NaN            24          3               231            548   
mean      50.372363           NaN        NaN               NaN            NaN   
std       28.821076           NaN        NaN               NaN            NaN   
min        1.000000           NaN        NaN               NaN            NaN   
25%       25.000000           NaN        NaN               NaN            NaN   
50%       50.000000           NaN        NaN               NaN            NaN   
75%       75.000000           NaN        NaN               NaN            NaN   
max      100.000000           NaN        NaN               NaN            NaN   

       worldwide_gross  
count             5782  
unique            5356  
top                 $0  
freq               367  
mean               NaN  
std                NaN  
min                NaN  
25%                NaN  
50%                NaN  
75%                NaN  
max                NaN

Data shows presence of NaN values (Not a Number) so they will need to be dealt with

In [66]:
#Getting the general description of the datasets to better understand the data

tn_movie_budgets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 1 columns):
 #   Column                                                                                             Non-Null Count  Dtype 
---  ------                                                                                             --------------  ----- 
 0   id	synopsis	rating	genre	director	writer	theater_date	dvd_date	currency	box_office	runtime	studio  1560 non-null   object
dtypes: object(1)
memory usage: 12.3+ KB


In [67]:
tmdb_movies.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26517 entries, 0 to 26516
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         26517 non-null  int64  
 1   genre_ids          26517 non-null  object 
 2   id                 26517 non-null  int64  
 3   original_language  26517 non-null  object 
 4   original_title     26517 non-null  object 
 5   popularity         26517 non-null  float64
 6   release_date       26517 non-null  object 
 7   title              26517 non-null  object 
 8   vote_average       26517 non-null  float64
 9   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(3), object(5)
memory usage: 2.0+ MB


In [68]:
bom_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [82]:
#Rows and columns of the data

print(bom_movies.shape)
print('The dataset has ' + str(bom_movies.shape[0]) + ' rows and ' + str(bom_movies.shape[1]) + ' columns')



(3387, 5)
The dataset has 3387 rows and 5 columns


In [83]:
print(tmdb_movies.shape)
print('The dataset has ' + str(tmdb_movies.shape[0]) + ' rows and ' + str(tmdb_movies.shape[1]) + ' columns')



(26517, 10)
The dataset has 26517 rows and 10 columns


In [84]:
print(tn_movie_budgets.shape)
print('The dataset has ' + str(tn_movie_budgets.shape[0]) + ' rows and ' + str(tn_movie_budgets.shape[1]) + ' columns')



(1560, 1)
The dataset has 1560 rows and 1 columns


In [90]:
#Read the heads and tails (preview) of the Datasets we'll work with.
bom_movies.head()
bom_movies.tail()



title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [91]:
tmdb_movies.head()



Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [92]:
tmdb_movies.tail()


Unnamed: 0        genre_ids      id original_language  \
26512       26512         [27, 18]  488143                en   
26513       26513         [18, 53]  485975                en   
26514       26514     [14, 28, 12]  381231                en   
26515       26515  [10751, 12, 28]  366854                en   
26516       26516         [53, 27]  309885                en   

              original_title  popularity release_date                  title  \
26512  Laboratory Conditions         0.6   2018-10-13  Laboratory Conditions   
26513        _EXHIBIT_84xxx_         0.6   2018-05-01        _EXHIBIT_84xxx_   
26514           The Last One         0.6   2018-10-01           The Last One   
26515           Trailer Made         0.6   2018-06-22           Trailer Made   
26516             The Church         0.6   2018-10-05             The Church   

       vote_average  vote_count  
26512           0.0           1  
26513           0.0           1  
26514           0.0           1  
26515           0.0           1  
26516           0.0           1

In [93]:
tn_movie_budgets.head()

id\tsynopsis\trating\tgenre\tdirector\twriter\ttheater_date\tdvd_date\tcurrency\tbox_office\truntime\tstudio
0  1\t"This gritty, fast-paced, and innovative po...                                                          
1  3\tNew York City, not-too-distant-future: Eric...                                                          
2  5\t"Illeana Douglas delivers a superb performa...                                                          
3  6\tMichael Douglas runs afoul of a treacherous...                                                          
4  7\t\tNR\tDrama|Romance\tRodney Bennett\tGiles ...

In [94]:
tn_movie_budgets.tail()

id\tsynopsis\trating\tgenre\tdirector\twriter\ttheater_date\tdvd_date\tcurrency\tbox_office\truntime\tstudio
1555  1996\tForget terrorists or hijackers -- there'...                                                          
1556  1997\tThe popular Saturday Night Live sketch w...                                                          
1557  1998\t"Based on a novel by Richard Powell, whe...                                                          
1558  1999\tThe Sandlot is a coming-of-age story abo...                                                          
1559  2000\tSuspended from the force, Paris cop Hube...

# Data Cleaning

Data cleaning helps us ensure data is clean by confirming that it is free of inaccuracies, inconcistencies, non-uniform and incompleteness of data.

In [8]:
#check for duplicates
bom_movies.duplicated().value_counts()


False    3387
dtype: int64

In [11]:
tmdb_movies.duplicated().value_counts()


False    26517
dtype: int64

In [12]:
tn_movie_budgets.duplicated().value_counts()


False    5782
dtype: int64

There are no duplicates recorded in either dataset, so we carry on

In [16]:
#Check for missing values
tmdb_movies.isna().any()


Unnamed: 0           False
genre_ids            False
id                   False
original_language    False
original_title       False
popularity           False
release_date         False
title                False
vote_average         False
vote_count           False
dtype: bool

In [18]:
tn_movie_budgets.isna().any()


id                   False
release_date         False
movie                False
production_budget    False
domestic_gross       False
worldwide_gross      False
dtype: bool

In [19]:
bom_movies.isna().any()


title             False
studio             True
domestic_gross     True
foreign_gross      True
year              False
dtype: bool

In [20]:
bom_movies.isna().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In [23]:
#Determine the percentage of rows of the foreign_gross column containing missing values
#To further determine what to do with the dataset

print('Percentage of Null foreign_gross Values:', len(bom_movies[bom_movies.foreign_gross.isna()])/ len(bom_movies))
print('Number of Unique foreign_gross Values:', bom_movies.foreign_gross.nunique())

Percentage of Null foreign_gross Values: 0.3985828166519043
Number of Unique foreign_gross Values: 1204


The bom_movies dataset has three columns with missing values. The studio and domestic gross column has a very low percentage of missing data that will not affect our analysis. The percentage of missing  values in the foreign gross column is 39.86%. This number is also not that high to warrant dropping the column. 

In [56]:
rt_movie = pd.read_csv('zippedData/rt.movie_info.tsv.gz', sep='\t')
rt_movie


id                                           synopsis rating  \
0        1  This gritty, fast-paced, and innovative police...      R   
1        3  New York City, not-too-distant-future: Eric Pa...      R   
2        5  Illeana Douglas delivers a superb performance ...      R   
3        6  Michael Douglas runs afoul of a treacherous su...      R   
4        7                                                NaN     NR   
...    ...                                                ...    ...   
1555  1996  Forget terrorists or hijackers -- there's a ha...      R   
1556  1997  The popular Saturday Night Live sketch was exp...     PG   
1557  1998  Based on a novel by Richard Powell, when the l...      G   
1558  1999  The Sandlot is a coming-of-age story about a g...     PG   
1559  2000  Suspended from the force, Paris cop Hubert is ...      R   

                                                  genre            director  \
0                   Action and Adventure|Classics|Drama    William Friedkin   
1                     Drama|Science Fiction and Fantasy    David Cronenberg   
2                     Drama|Musical and Performing Arts      Allison Anders   
3                            Drama|Mystery and Suspense      Barry Levinson   
4                                         Drama|Romance      Rodney Bennett   
...                                                 ...                 ...   
1555   Action and Adventure|Horror|Mystery and Suspense                 NaN   
1556                 Comedy|Science Fiction and Fantasy        Steve Barron   
1557  Classics|Comedy|Drama|Musical and Performing Arts      Gordon Douglas   
1558    Comedy|Drama|Kids and Family|Sports and Fitness  David Mickey Evans   
1559  Action and Adventure|Art House and Internation...                 NaN   

                                                writer  theater_date  \
0                                       Ernest Tidyman   Oct 9, 1971   
1                         David Cronenberg|Don DeLillo  Aug 17, 2012   
2                                       Allison Anders  Sep 13, 1996   
3                      Paul Attanasio|Michael Crichton   Dec 9, 1994   
4                                         Giles Cooper           NaN   
...                                                ...           ...   
1555                                               NaN  Aug 18, 2006   
1556  Terry Turner|Tom Davis|Dan Aykroyd|Bonnie Turner  Jul 23, 1993   
1557                                               NaN   Jan 1, 1962   
1558                  David Mickey Evans|Robert Gunter   Apr 1, 1993   
1559                                        Luc Besson  Sep 27, 2001   

          dvd_date currency  box_office      runtime             studio  
0     Sep 25, 2001      NaN         NaN  104 minutes                NaN  
1      Jan 1, 2013        $     600,000  108 minutes  Entertainment One  
2     Apr 18, 2000      NaN         NaN  116 minutes                NaN  
3     Aug 27, 1997      NaN         NaN  128 minutes                NaN  
4              NaN      NaN         NaN  200 minutes                NaN  
...            ...      ...         ...          ...                ...  
1555   Jan 2, 2007        $  33,886,034  106 minutes    New Line Cinema  
1556  Apr 17, 2001      NaN         NaN   88 minutes  Paramount Vantage  
1557  May 11, 2004      NaN         NaN  111 minutes                NaN  
1558  Jan 29, 2002      NaN         NaN  101 minutes                NaN  
1559  Feb 11, 2003      NaN         NaN   94 minutes  Columbia Pictures  

[1560 rows x 12 columns]

In [34]:
rt_movie.isnull().sum()

id                 0
synopsis          62
rating             3
genre              8
director         199
writer           449
theater_date     359
dvd_date         359
currency        1220
box_office      1220
runtime           30
studio          1066
dtype: int64

In [57]:
rt_reviews = pd.read_csv('zippedData/rt.reviews.tsv.gz', sep='\t', encoding='unicode_escape')
rt_reviews

id                                             review rating   fresh  \
0         3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1         3  It's an allegory in search of a meaning that n...    NaN  rotten   
2         3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3         3  Continuing along a line introduced in last yea...    NaN   fresh   
4         3             ... a perverse twist on neorealism...     NaN   fresh   
...     ...                                                ...    ...     ...   
54427  2000  The real charm of this trifle is the deadpan c...    NaN   fresh   
54428  2000                                                NaN    1/5  rotten   
54429  2000                                                NaN    2/5  rotten   
54430  2000                                                NaN  2.5/5  rotten   
54431  2000                                                NaN    3/5   fresh   

                  critic  top_critic         publisher                date  
0             PJ Nabarro           0   Patrick Nabarro   November 10, 2018  
1         Annalee Newitz           0           io9.com        May 23, 2018  
2           Sean Axmaker           0  Stream on Demand     January 4, 2018  
3          Daniel Kasman           0              MUBI   November 16, 2017  
4                    NaN           0      Cinema Scope    October 12, 2017  
...                  ...         ...               ...                 ...  
54427      Laura Sinagra           1     Village Voice  September 24, 2002  
54428  Michael Szymanski           0        Zap2it.com  September 21, 2005  
54429       Emanuel Levy           0   EmanuelLevy.Com       July 17, 2005  
54430   Christopher Null           0    Filmcritic.com   September 7, 2003  
54431    Nicolas Lacroix           0      Showbizz.net   November 12, 2002  

[54432 rows x 8 columns]

In [72]:
#Merging the 4 dataframes 
#Resetting the DF name for brevity and to preserve orignial df 

df1 = pd.merge(bom_movies, tmdb_movies, on="title", how='inner')
df2 = pd.merge(df1, tn_movie_budgets, on="id", how='inner')  
df3 = pd.merge(df2, rt_movie, on="id", how='inner')  
#df = pd.concat([df3, rt_reviews], ignore_index=True, sort=False)
df = pd.merge(df3, rt_reviews, on="id", how='inner') 
df


Empty DataFrame
Columns: [title, studio_x, domestic_gross_x, foreign_gross, year, Unnamed: 0, genre_ids, original_language, original_title, popularity, release_date_x, vote_average, vote_count, release_date_y, movie, production_budget, domestic_gross_y, worldwide_gross, synopsis, rating_x, genre, director, writer, theater_date, dvd_date, currency, box_office, runtime, studio_y, id, review, rating_y, fresh, critic, top_critic, publisher, date]
Index: []

[0 rows x 37 columns]

In [69]:
df.head()

title studio_x  domestic_gross_x  \
0                                  Toy Story 3       BV       415000000.0   
1                   Alice in Wonderland (2010)       BV       334200000.0   
2  Harry Potter and the Deathly Hallows Part 1       WB       296000000.0   
3                   Clash of the Titans (2010)       WB       163200000.0   
4                                  Tron Legacy       BV       172100000.0   

  foreign_gross    year  Unnamed: 0        genre_ids       id  \
0     652000000  2010.0         7.0  [16, 10751, 35]  10193.0   
1     691300000  2010.0         NaN              NaN      NaN   
2     664300000  2010.0         NaN              NaN      NaN   
3     330000000  2010.0         NaN              NaN      NaN   
4     228000000  2010.0         NaN              NaN      NaN   

  original_language original_title  ...  box_office runtime  studio_y  review  \
0                en    Toy Story 3  ...         NaN     NaN       NaN     NaN   
1               NaN            NaN  ...         NaN     NaN       NaN     NaN   
2               NaN            NaN  ...         NaN     NaN       NaN     NaN   
3               NaN            NaN  ...         NaN     NaN       NaN     NaN   
4               NaN            NaN  ...         NaN     NaN       NaN     NaN   

  rating_y fresh critic top_critic publisher date  
0      NaN   NaN    NaN        NaN       NaN  NaN  
1      NaN   NaN    NaN        NaN       NaN  NaN  
2      NaN   NaN    NaN        NaN       NaN  NaN  
3      NaN   NaN    NaN        NaN       NaN  NaN  
4      NaN   NaN    NaN        NaN       NaN  NaN  

[5 rows x 37 columns]

In [70]:
df.shape

(266778, 37)

In [71]:
df.isna().sum()

title                229232
studio_x             262832
domestic_gross_x     262859
foreign_gross        264356
year                 262827
Unnamed: 0           230318
genre_ids            230318
id                     1086
original_language    230318
original_title       230318
popularity           230318
release_date_x       230318
vote_average         230318
vote_count           230318
release_date_y        79364
movie                 79364
production_budget     79364
domestic_gross_y      79364
worldwide_gross       79364
synopsis              33864
rating_x              33659
genre                 33664
director              73958
writer                91920
theater_date          40348
dvd_date              40348
currency             124312
box_office           124312
runtime               34541
studio_y              95115
review                57871
rating_y              86559
fresh                 30342
critic                41316
top_critic            30342
publisher           

# Data Analysis

# # Question 1: Are movies profitable?

In [6]:
#create a new dataframe that will isolate other variables but tittle and domestic gross

profitability = df.drop(["studio", "foreign_gross", "year",
                         "Unnamed: 0", "genre_ids", "id", "original_language", "original_title", "popularity",
                         "release_date", "vote_average", "vote_count", "movie", "production_budget", "worldwide_gross"], axis =1)
profitability

title domestic_gross
0                                      Toy Story 3    415000000.0
1                       Alice in Wonderland (2010)    334200000.0
2      Harry Potter and the Deathly Hallows Part 1    296000000.0
3                                        Inception    292600000.0
4                              Shrek Forever After    238700000.0
...                                            ...            ...
33380                                          NaN             $0
33381                                          NaN        $48,482
33382                                          NaN         $1,338
33383                                          NaN             $0
33384                                          NaN       $181,041

[33385 rows x 2 columns]

In [55]:
#NaN values
profitability.isnull().sum()

title              5782
domestic_gross    23846
dtype: int64

In [57]:
print('Percentage of Null foreign_gross Values:', len(profitability[profitability.title.isna()])/ len(profitability))
print('Percentage of Null foreign_gross Values:', len(profitability[profitability.domestic_gross.isna()])/ len(profitability))


Percentage of Null foreign_gross Values: 0.1731915530927063
Percentage of Null foreign_gross Values: 0.7142728770405871


In [59]:
#drop the title rows with NaN values
profitability.dropna(subset= ['title'], inplace= True)
profitability.isnull().sum()

title                 0
domestic_gross    23846
dtype: int64

In [60]:
#Calculating domestic gross mean in dollars
profitability['domestic_gross'].mean() 


29993444.20415225

In [64]:
#Calculating domestic gross median in dollars

profitability['domestic_gross'].median() 
a.merr(a,)

1500000.0

In [65]:
#Replace the NaN values in the domestic gross column with the domestic gross median
profitability.fillna(1500000.0)


title  domestic_gross
0                                      Toy Story 3     415000000.0
1                       Alice in Wonderland (2010)     334200000.0
2      Harry Potter and the Deathly Hallows Part 1     296000000.0
3                                        Inception     292600000.0
4                              Shrek Forever After     238700000.0
...                                            ...             ...
27598                        Laboratory Conditions       1500000.0
27599                              _EXHIBIT_84xxx_       1500000.0
27600                                 The Last One       1500000.0
27601                                 Trailer Made       1500000.0
27602                                   The Church       1500000.0

[27603 rows x 2 columns]

In [8]:
#Plot the data to visualize title vs domestic gross

ax = sns.barplot(x="title", y="domestic_gross", data=profitability)  


TypeError: Neither the `x` nor `y` variable appears to be numeric.

# #Question 2: Which genre has the highest ratings? 

In [17]:
#create a df to answer the question
#drop all columns not needed to answer the question

ratings_genre = df.drop(["studio", "foreign_gross", "year",
                         "Unnamed: 0", "genre_ids", "id", "original_language", "original_title", "popularity",
                         "release_date", "vote_average", "vote_count", "movie", "production_budget", "worldwide_gross",
                        "domestic_gross", "synopsis", "title", "director", "writer", "theater_date", "dvd_date","currency",
                         "box_office", "runtime"], axis =1)
ratings_genre

rating                                              genre
0        NaN                                                NaN
1        NaN                                                NaN
2        NaN                                                NaN
3        NaN                                                NaN
4        NaN                                                NaN
...      ...                                                ...
34940      R   Action and Adventure|Horror|Mystery and Suspense
34941     PG                 Comedy|Science Fiction and Fantasy
34942      G  Classics|Comedy|Drama|Musical and Performing Arts
34943     PG    Comedy|Drama|Kids and Family|Sports and Fitness
34944      R  Action and Adventure|Art House and Internation...

[34945 rows x 2 columns]

In [18]:
ratings_genre.isnull().sum()

rating    33388
genre     33393
dtype: int64

In [19]:
print('Percentage of Null genre values:', len(ratings_genre[ratings_genre.genre.isna()])/ len(ratings_genre))
print('Percentage of Null rating values:', len(ratings_genre[ratings_genre.rating.isna()])/ len(ratings_genre))



Percentage of Null genre values: 0.9555873515524396
Percentage of Null rating values: 0.9554442695664616


id                                             review rating   fresh  \
0         3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1         3  It's an allegory in search of a meaning that n...    NaN  rotten   
2         3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3         3  Continuing along a line introduced in last yea...    NaN   fresh   
4         3             ... a perverse twist on neorealism...     NaN   fresh   
...     ...                                                ...    ...     ...   
54427  2000  The real charm of this trifle is the deadpan c...    NaN   fresh   
54428  2000                                                NaN    1/5  rotten   
54429  2000                                                NaN    2/5  rotten   
54430  2000                                                NaN  2.5/5  rotten   
54431  2000                                                NaN    3/5   fresh   

                  critic  top_critic         publisher                date  
0             PJ Nabarro           0   Patrick Nabarro   November 10, 2018  
1         Annalee Newitz           0           io9.com        May 23, 2018  
2           Sean Axmaker           0  Stream on Demand     January 4, 2018  
3          Daniel Kasman           0              MUBI   November 16, 2017  
4                    NaN           0      Cinema Scope    October 12, 2017  
...                  ...         ...               ...                 ...  
54427      Laura Sinagra           1     Village Voice  September 24, 2002  
54428  Michael Szymanski           0        Zap2it.com  September 21, 2005  
54429       Emanuel Levy           0   EmanuelLevy.Com       July 17, 2005  
54430   Christopher Null           0    Filmcritic.com   September 7, 2003  
54431    Nicolas Lacroix           0      Showbizz.net   November 12, 2002  

[54432 rows x 8 columns]

In [28]:
sf.isnull().sum()

id                0
review         5563
rating        13517
fresh             0
critic         2722
top_critic        0
publisher       309
date              0
dtype: int64